routes/user/Auth.js

```diff
@@ -156,16 +157,38 @@ router.get(
   passport.authenticate("google", {
     failureRedirect: "auth/login/failed",
   }),
-  (req, res) => {
-    res
-      .cookie("OAuthLoginInitiated", true, {
-        expires: new Date(new Date().getTime() + 5 * 60 * 1000),
-        httpOnly: false,
-        secure: true,
-        sameSite: "none",
-        domain: process.env.ORIGIN_DOMAIN,
-      })
-      .redirect(process.env.successURL);
+  async (req, res) => {
+    const usertype = req.query.state;
+    if (req.isAuthenticated()) {
+      const user = req.user;
+      try {
+        const updatedUser = await User.findOneAndUpdate(
+          { email: user.email },
+          { usertype },
+          { new: true },
+        );
+        if (!updatedUser) {
+          return res.status(STATUSCODE.INTERNAL_SERVER_ERROR).json({
+            message: "Failed to update user profile",
+          });
+        }
+        res
+          .cookie("OAuthLoginInitiated", true, {
+            expires: new Date(new Date().getTime() + 5 * 60 * 1000),
+            httpOnly: false,
+            secure: true,
+            sameSite: "none",
+            domain: process.env.ORIGIN_DOMAIN,
+          })
+          .redirect(process.env.successURL);
+      } catch (err) {
+        res.status(STATUSCODE.INTERNAL_SERVER_ERROR).json({ message: err });
+      }
+    } else {
+      res
+        .status(STATUSCODE.UNAUTHORIZED)
+        .json({ error: true, message: STATUSMESSAGE.UNAUTHORIZED });
+    }
```

Comment:
Can you please explain what exactly are you trying to do in this part ? Would like to know more about the thought process.

Reply:
Basically Google Oauth doesn't allow extra custom parameter in the during Authentication dialog box. So we are taking the usertype from the user first then passing that parameter in backend so as soon as user login or signup with google he\her usertype will be set to the desired type. Prefer the video link:

https://www.loom.com/share/5af1c8739e984f5ab33beef2b3da302c?sid=f437fd6a-0ee9-4b89-92eb-6f3add8d679e

Reply:
I checked out the video and would like to suggest a few things, and questions.

 - The frontend route would be `/user/username` for users and `/club/username` for clubs. Not individual.
 - What is the use of the update ? If I can choose between making a user or a club, why are we updating the details ?
```js
const updatedUser = await User.findOneAndUpdate(
          { email: user.email },
          { usertype },
          { new: true },
        );
        if (!updatedUser) {
          return res.status(STATUSCODE.INTERNAL_SERVER_ERROR).json({
            message: "Failed to update user profile",
          });
        }
```
- I saw that you made some UI changes in the Frontend too however there can be some more changes, and we can talk about those later.
- I have a question that is if I do not have an account and I click on `continue with Google` what happens in that scenario ? Do I get to choose and then a new account with the usertype that I chose is made ?

Reply:
- I have fixed the route to `/user/username` and `/club/clubname` as you have mentioned.
- I have removed updateUser part as it was setting the type of user different everytime. I have made a small change there
- Yes I will create a pull request in frontend after this.
- For now both sign-up and sign-in is same in frontend but if user does not exist then it'll set the usertype. If exist then it won't change the usertype.

Reply:
So let me get this through real quick, if I don't have an account and I continue with Google a new account will be created in the DB, along with the `usertype` of my choice ?

And if i do have, I simply just log in and it works as usual. Right ?

Reply:
Yes exactly.. If you don't have any account it will create your account along with userType.. if you have account it's just simple login

Reply:
Cool, that works !